gene_finder.py

```diff
@@ -22,145 +22,238 @@ def shuffle_string(s):
+                return result
+        return dna
+
+    def find_all_ORFs_oneframe(dna):
+        """
+        Finds all non-nested open reading frames in the given DNA
         sequence and returns them as a list.  This function should
         only find ORFs that are in the default frame of the sequence
         (i.e. they start on indices that are multiples of 3).
         By non-nested we mean that if an ORF occurs entirely within
         another ORF, it should not be included in the returned list of ORFs.
-
         dna: a DNA sequence
         returns: a list of non-nested ORFs
-    >>> find_all_ORFs_oneframe("ATGCATGAATGTAGATAGATGTGCCC")
-    ['ATGCATGAATGTAGA', 'ATGTGCCC']
-    """
-    # TODO: implement this
-    pass
-
-
-def find_all_ORFs(dna):
-    """ Finds all non-nested open reading frames in the given DNA sequence in
+        >>> find_all_ORFs_oneframe("ATGCATGAATGTAGATAGATGTGCCC")
+        ['ATGCATGAATGTAGA', 'ATGTGCCC']
+        >>> find_all_ORFs_oneframe("ATGTGAA")
+        ['ATG']
+        >>> find_all_ORFs_oneframe('ASDASDAWSDSD')
+        []
+        >>> find_all_ORFs_oneframe('TATATGCATGAATGTAGATAGATGTGCTAAATAATAATGTTTTAAATT')
+        ['ATGCATGAATGTAGA', 'ATGTGC', 'ATGTTT']
+        """
+        index = 0
+        orf_list = []
+        while index < len(dna):
+            if dna[index:index+3] == 'ATG':
+                #appended ORF
+                orf = rest_of_ORF(dna[index:])
+                orf_list.append(orf)
+                index = index + len(orf)
+            else:
+                index = index + 3
+        return orf_list
+
+    def find_all_ORFs(dna):
+        """
+        Finds all non-nested open reading frames in the given DNA sequence in
         all 3 possible frames and returns them as a list.  By non-nested we
         mean that if an ORF occurs entirely within another ORF and they are
         both in the same frame, it should not be included in the returned list
         of ORFs.
-
         dna: a DNA sequence
         returns: a list of non-nested ORFs
-
-    >>> find_all_ORFs("ATGCATGAATGTAG")
-    ['ATGCATGAATGTAG', 'ATGAATGTAG', 'ATG']
-    """
-    # TODO: implement this
-    pass
-
-
-def find_all_ORFs_both_strands(dna):
-    """ Finds all non-nested open reading frames in the given DNA sequence on both
+        This unit testing would be enough because there isn't any special exceptions that needs to be tested. Also, this case tests this function's
+        ability to grab orf from three different possible reading frames.
+        >>> find_all_ORFs("ATGCATGAATGTAG")
+        ['ATGCATGAATGTAG', 'ATGAATGTAG', 'ATG']
+        """
+        #orf list from all frames
+        orf_list = []
+        #zero offset frame
+        orf_list = orf_list + find_all_ORFs_oneframe(dna)
+        #first offset frame
+        orf_list = orf_list + find_all_ORFs_oneframe(dna[1:])
+        #second offset frame
+        orf_list = orf_list + find_all_ORFs_oneframe(dna[2:])
+        return orf_list
```

Comment:
If frames were larger than 3 codons long, you would have to copy the same line over and over, so putting it in a loop is a good choice.